scalding-quotation/src/test/scala/com/twitter/scalding/quotation/LimitationsTest.scala

```diff
@@ -0,0 +1,25 @@
+package com.twitter.scalding.quotation
+
+class LimitationsTest extends Test {
+
+  class TestClass {
+    def function[T, U](f: T => U)(implicit q: Quoted) = (q, f)
+  }
+
+  val test = new TestClass
+
+  "nested transitive projection" in pendingUntilFixed {
+    test.function[Person, Option[String]](_.alternativeContact.map(_.phone))._1.projections.set mustEqual
+      Set(Person.typeReference.andThen(Accessor("alternativeContact"), typeName[Option[Contact]]).andThen(Accessor("phone"), typeName[String]))
```

Comment:
this confuses me. How do we distinguish an `andThen` projection inside or outside an `Option`? Seems like we need to record the input type we are dealing with also. We have `Option[Contact]` as the output of the previous Projection, but how do we jump across the `.map` on the Option?

Reply:
Note that this is a failing test with `pendingUntilFixed`. The projections in this scenario are `Person.alternativeContact` and `Contact.phone`, which are unrelated and thus don't get filtered out by `Projections`. Ideally, the macro should apply a transformation similar to beta reduction to produce only the projection `Person.alternativeContact.phone`.